packages/api/app/Models/Token.js

```diff
@@ -47,6 +50,21 @@ class Token extends Model {
 			)
 			.first();
 	}
+
+	static async generateUniqueTokenCode() {
+		let uniqueCode = false;
+		let generatedCode = null;
+		let existentCode = null;
+		do {
+			const minm = 100000;
+			const maxm = 999999;
+			generatedCode = Math.floor(Math.random() * (maxm - minm + 1)) + minm;
+			// eslint-disable-next-line no-await-in-loop
+			existentCode = await this.findBy('token', generatedCode);
+			if (!existentCode) uniqueCode = true;
+		} while (!uniqueCode);
+		return generatedCode;
+	}
```

Comment:
O que acha da implementação abaixo? Creio que ficou mais fácil de entender e, mudando o min/max de lugar evitamos recriá-los a cada iteração 
```suggestion
	static async generateUniqueTokenCode() {
		const min = 100000;
		const max = 999999;
		
		while(true) {
			const generatedCode = Math.floor(Math.random() * (max - min + 1)) + min;
			// eslint-disable-next-line no-await-in-loop
			if (!await this.findBy('token', generatedCode)) {
				return generatedCode;
			}
		}
	}
```

Reply:
> O que acha da implementação abaixo? Creio que ficou mais fácil de entender e, mudando o min/max de lugar evitamos recriá-los a cada iteração

Dá certo!